### ISSUE TITLE
Application `:comeonin` is required

### ISSUE THREAD

**Issue by pablobm** [BlakeWilliams/doorman#27]
We just had a situation where our staging environment was denying that `Comeonin` was loaded at all, as follows:

```
** (exit) an exception was raised:
    ** (UndefinedFunctionError) function Comeonin.Bcrypt.hashpwsalt/1 is undefined (module Comeonin.Bcrypt is not available)
        Comeonin.Bcrypt.hashpwsalt("password123")
        (doorman) lib/auth/bcrypt.ex:45: Doorman.Auth.Bcrypt.hash_password/1
        (lightswap) lib/lightswap/users/users.ex:54: Lightswap.Users.create_user/1
        (lightswap) lib/lightswap_web/controllers/user_controller.ex:14: LightswapWeb.UserController.create/2
        (lightswap) lib/lightswap_web/controllers/user_controller.ex:1: LightswapWeb.UserController.action/2
        (lightswap) lib/lightswap_web/controllers/user_controller.ex:1: LightswapWeb.UserController.phoenix_controller_pipeline/2
        (lightswap) lib/lightswap_web/endpoint.ex:1: LightswapWeb.Endpoint.instrument/4
        (phoenix) lib/phoenix/router.ex:278: Phoenix.Router.__call__/1
```

 After examining the build logs, we saw that we needed to explicitly include the application `:comeonin` in ours. See:

```
==> One or more direct or transitive dependencies are missing from
    :applications or :included_applications, they will not be included
    in the release:

    :comeonin

    This can cause your application to fail at runtime. If you are sure
    that this is not an issue, you may ignore this warning.
```

This was on Phoenix 1.3. Could that be the reason of this change? Also: why on staging (a virtual host running Ubuntu Xenial) but not on development (macOS Sierra)?

I'm not sure what the correct fix here is: whether to add the detail above to Doorman's documentation, to add this to its list of applications, or something else?

**Comment by BlakeWilliams** [comment]
Great question, I think you might be right about the list of applications. Can you try modifying this projects `mix.exs` file and add `:comeonin` to `applications` next to logger here: https://github.com/BlakeWilliams/doorman/blob/master/mix.exs#L20?

I think you can change it locally if you edit the file in `_deps` then `rm -rf _build` and re-run the application.

**Comment by pablobm** [comment]
Oh, forgot to mention: we indeed solved the problem doing exactly that. Now we are wondering if this should be in Doorman's documentation (for users to include the application manually), or Doorman can solve it itself.

**Comment by BlakeWilliams** [comment]
Oh interesting. I think it makes sense for Doorman to handle that. I can't imagine why not.

I'd happily accept a PR for that. If not I can get to it sometime this week or the next.

**Comment by pablobm** [comment]
I would volunteer for this... but my Elixir-fu is a bit lacking, so I wouldn't know where to start. Also I won't have time for at least two weeks. If after that time this is still open, I may give it a go.

**Comment by BlakeWilliams** [comment]
Closed via #30
